### DISCUSSION TITLE
How to avoid critical vulnerabilities?

### DISCUSSION THREAD

**Discussion by Andr3asB** [urlaubsverwaltung/zeiterfassung#267] [Q&A] [answered]
I was about to install your docker container for some testing of your solution for a demo with one of our customers. The customer would subscribe to the paid version later.
When downloading the docker image Docker desktop reports about 51 vulnerabilities whereof 8 are critical (CVSS 9.8 and 10)!

What is your recommendation to avoid these issues when implementing Zeiterfassung in an enterprise environment?

![image](https://github.com/urlaubsverwaltung/zeiterfassung/assets/44557554/6e90e944-6676-491a-8cc1-4df6abcfe46f)

**Comment by derTobsch** [comment, marked as answer]
Hallo @Andr3asB,

wir updaten die dependencies der Zeiterfassung ständig und veröffentlichen in regelmäßigen Abständen. Dadurch werden die Vulnerabilities zeitnah behoben. Es werden jedoch immer wieder neue auftreten, welche mit der nächsten Version vermutlich behoben werden. Wir können Vulnerabilities in einer Abhängigkeit, welche wir einsetzen, nicht direkt beheben, sind aber, wie gesagt, daran interessiert zeitnah eine neue Version zur Verfügung zu stellen.

Kritische Vulnerabilities, welche zum Beispiel eine "remote code execution" ermöglichen oder als sehr hoch eingestuft werden, werden von uns direkt behoben und auf den System ausgerollt.
